crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2312,7 +2312,7 @@ impl Client {
         });
         let mut request_config = self.request_config();
         if let Some(timeout) = sync_settings.timeout {
-            request_config.timeout = Some(timeout);
+            request_config.timeout = Some(request_config.timeout.unwrap_or_default() + timeout);
```

Comment:
Hmm, the default here would be 0, let's `unwrap()` into a `Duration` of 30s instead.

Sorry I didn't realize that the default value wouldn't work either in my first comment.